From 24a6efeea14b619eca47a27fbcbc35e306f81a55 Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Fri, 5 Sep 2014 11:03:37 -0400
Subject: [ethernet] sfc: calls skb_set_hash

Message-id: <1409915048-26438-2-git-send-email-nikolay@redhat.com>
Patchwork-id: 91398
O-Subject: [RHEL7.1 PATCH BZ 1110888 01/32] net: sfc calls skb_set_hash
Bugzilla: 1110888
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Tom Herbert <therbert@google.com>

Drivers should call skb_set_hash to set the hash and its type
in an skbuff.

Signed-off-by: Tom Herbert <therbert@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c7cb38af79c1e71a72d398f39296f1e27779b778)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/sfc/rx.c b/drivers/net/ethernet/sfc/rx.c
index 1fde9b8..48588dd 100644
--- a/drivers/net/ethernet/sfc/rx.c
+++ b/drivers/net/ethernet/sfc/rx.c
@@ -440,7 +440,8 @@ efx_rx_packet_gro(struct efx_channel *channel, struct efx_rx_buffer *rx_buf,
  }
 
  if (efx->net_dev->features & NETIF_F_RXHASH)
-  skb->rxhash = efx_rx_buf_hash(efx, eh);
+  skb_set_hash(skb, efx_rx_buf_hash(efx, eh),
+        PKT_HASH_TYPE_L3);
  skb->ip_summed = ((rx_buf->flags & EFX_RX_PKT_CSUMMED) ?
      CHECKSUM_UNNECESSARY : CHECKSUM_NONE);
 
-- 
1.7.1